Azure/azure-cli
# [ARM] az deployment: Fix the problem of incorrect message of JSON after removing comments and compressing (#13159)

Issue: #13159

**Description**  
When the JSON format of the template file or parameter file is not correct, the following error will be reported:
```pwsh
Expecting value: line 1 column 337 (char 336)
```
There are two problems with it:
1. Because the processing of removing comments and compression in the method `_remove_comments_from_json` will lead to misplacement of error location in error message, so the error message should be wrapped.
2. It does not prompt the specific error file. It may be a template file or a parameter file.

**Testing Guide**  
Deploy or verify the template file or parameter file with incorrect JSON format.


**History Notes**  


[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
